ibis/backends/pyspark/__init__.py

```diff
@@ -929,3 +940,110 @@ def to_pyarrow_batches(
         return pa.ipc.RecordBatchReader.from_batches(
             pa_table.schema, pa_table.to_batches(max_chunksize=chunk_size)
         )
+
+    @util.experimental
+    def read_kafka(
+        self,
+        table_name: str | None = None,
+        watermark: Watermark | None = None,
+        auto_parse: bool = False,
+        schema: sch.Schema | None = None,
+        options: Mapping[str, str] | None = None,
+    ) -> ir.Table:
+        """Register a Kafka topic as a table in the current database.
+
+        Parameters
+        ----------
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+        watermark
+            Watermark strategy for the table.
+        auto_parse
+            Whether to parse Kafka messages automatically. If `False`, the source is read
+            as binary keys and values. If `True`, the key is discarded and the value is
+            parsed using the provided schema.
+        schema
+            Schema of the value of the Kafka messages.
+        options
+            Additional arguments passed to PySpark as .option("key", "value").
```

Comment:
From the code it seems like these are Kafka options, by way of PySpark, but they're not options for PySpark. Is that true? If so, this description should be a more clear that it's KafKa optionality being affected here. Alternatively, `Additional arguments passe to Kafka via PySpark` would also be fine.

Reply:
What do you mean by "Kafka options" but not "options for PySpark"? On the URL that I linked in the line below https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html it lists out all the configurations that Pyspark are required vs supported when connecting to Kafka sink/source from Pyspark, and it doesn't explicitly mention that they're passed to Kafka directly. They're not PySpark session configurations, but they are configurations that PySpark manages

Reply:
Ah, I think I am thrown off by the fact that they are clearly only related to Kafka, but being specified in the PySpark API.

What you have here is fine as is then!